restapi/src/main/java/io/stargate/web/docsapi/service/QueryExecutor.java

```diff
@@ -54,6 +54,12 @@ public QueryExecutor(DataStore dataStore) {
       int pageSize,
       ByteBuffer pagingState,
       ExecutionContext context) {
+    if (pageSize <= 0) {
+      // Note: if page size is not set, C* will ignore paging state, but subsequent pages may be
+      // requested by the execute() method, so we require a specific page size for all queries here.
+      throw new IllegalArgumentException("Unsupported page size: " + pageSize);
```

Comment:
can we map this to our new error code exception?

Reply:
This is not a user-level page size. So if that fails (unexpected) it would be a `500` on the HTTP size. Does it make sense to have a dedicated error code enum in this case?